src/main/java/org/broadinstitute/consent/http/resources/DarCollectionResource.java

```diff
@@ -96,6 +96,47 @@ public Response getCollectionSummariesForUserByRole(@Auth AuthUser authUser, @Pa
     }
   }
 
+
+  @GET
+  @Path("role/{roleName}/summary/{collectionId}")
+  @Produces("application/json")
+  @RolesAllowed({ADMIN, CHAIRPERSON, MEMBER, SIGNINGOFFICIAL, RESEARCHER})
+  public Response getCollectionSummaryForRoleById(@Auth AuthUser authUser, @PathParam("roleName") String roleName, @PathParam("collectionId") Integer collectionId) {
+    try {
+      User user = userService.findUserByEmail(authUser.getEmail());
+      validateUserHasRoleName(user, roleName); //throws BadRequestException if user does not have roleName
+      DarCollection collection = darCollectionService.getByCollectionId(collectionId);
+
+      boolean allowedAccess;
+      switch (roleName) {
+        case Resource.ADMIN:
+          allowedAccess = true;
+          break;
+        case Resource.CHAIRPERSON:
+        case Resource.MEMBER:
+          allowedAccess = checkDacPermissionsForCollection(user, collection);
+          break;
+        case Resource.SIGNINGOFFICIAL:
+          allowedAccess = checkSoPermissionsForCollection(user, collection);
+          break;
+        case Resource.RESEARCHER:
+          allowedAccess = user.getUserId().equals(collection.getCreateUserId());
+          break;
+        default:
+          throw new BadRequestException("Invalid role selection: " + roleName);
+      }
+      if (!allowedAccess) {
+        // user has role but is not allowed to view collection; throw NotFoundException to avoid leaking existence
+        throw new NotFoundException("Collection with the collection id of " + collectionId + " was not found");
+      }
+
+      DarCollectionSummary summary = darCollectionService.getSummaryForRoleNameByCollectionId(user, roleName, collectionId);
```

Comment:
Optional idea here. We're querying for the collection twice, here and on line 108. If there were methods for checking `allowedAccess`  on a summary instead of a collection, you could reduce the work required to complete this task.

Reply:
We can do that. It would involve adding the `institutionId` and the `userId` of the researcher to the summary queries and then copying the logic in existing helper methods (`checkDacPermissionsForCollection`) to determine `allowedAccess`. Would it be better to include the institutionId and userId in the previous summary queries so that these properties aren't null, or to leave them alone since it's not necessary for the consoles?